src/Listeners/UpdateSyncedResource.php

```diff
@@ -112,11 +113,24 @@ protected function updateResourceInTenantDatabases($tenants, $event, $syncedAttr
                     $localModel->update($syncedAttributes);
                 } else {
                     // When creating, we use all columns, not just the synced ones.
-                    $localModel = $localModelClass::create($eventModel->getAttributes());
+                    $localModel = $localModelClass::create($this->getResourceCreationAttributes($eventModel));
                 }
 
                 event(new SyncedResourceChangedInForeignDatabase($localModel, $tenant));
             });
         });
     }
+
+    protected function getResourceCreationAttributes(Syncable $model): array
```

Comment:
I'd name this something else than the method on the model. The code is now a bit confusing, seeing `$this->getResourceCreationAttributes($eventModel)` isn't immediately clear that it's something on this class rather than on the model.

Perhaps rename this to `getAttributesForCreation($model)`.

FWIW I don't fully like the names of either of these two methods right now but I can't think of anything better, so I'll rename them later.

Reply:
For now, renaming the method to `getAttributesForCreation`.